src-cpp/rdkafkacpp.h

```diff
@@ -146,6 +146,40 @@ std::string get_debug_contexts();
 RD_EXPORT
 int          wait_destroyed(int timeout_ms);
 
+/**
+ * @brief Allocate memory using the same allocator librdkafka uses.
+ *
+ * This is typically an abstraction for the malloc(3) call and makes sure
+ * the application can use the same memory allocator as librdkafka for
+ * allocating pointers that are used by librdkafka.
+ *
+ * \p handle of type rd_kafka_t can be set to return memory allocated by a specific \c handle instance
```

Comment:
Why pass the C handle and not a C++ handle?

Reply:
There is no C++ handle for rd_kafka_t, or maybe I missed it.

Reply:
The RdKafka::Handle has a rk_ pointer.

Reply:
Ok, replacing with RdKafka::Handle instance